x-pack/platform/plugins/shared/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -131,16 +132,24 @@ export const getAgentPoliciesHandler: FleetRequestHandler<
   TypeOf<typeof GetAgentPoliciesRequestSchema.query>
 > = async (context, request, response) => {
   const [coreContext, fleetContext] = await Promise.all([context.core, context.fleet]);
-  const soClient = fleetContext.internalSoClient;
+
+  const authzFleetReadAgentPolicies =
+    request.authzResult?.[FLEET_API_PRIVILEGES.AGENT_POLICIES.READ] === true;
+  const authzFleetAgentRead = request.authzResult?.[FLEET_API_PRIVILEGES.AGENTS.READ] === true;
+
+  const soClient = authzFleetReadAgentPolicies
+    ? coreContext.savedObjects.client
+    : fleetContext.internalSoClient;
```

Comment:
if `authzFleetReadAgentPolicies` is false, isn't the logic return results from all spaces with `internalSoClient`?

Reply:
yes in the case the user is accessing fleet with a permissions like setup or endpoint, we need an internal SO client

Reply:
shouldn't we explicitly check those permissions? what happens if the user doesn't have any permissions to fleet?

Reply:
Actually it seems for user with fleet-setup permissions (old fleet server) we need to user an internalSoClient, for the rest of the calls we could use a scopped SO client, I will update the PR accordingly